xbmc/events/BaseEvent.cpp

```diff
@@ -97,3 +104,13 @@ std::string CBaseEvent::VariantToLocalizedString(const CVariant& variant)
 
   return "";
 }
+
+void CBaseEvent::ToSortable(SortItem& sortable, Field field) const
+{
+  if (field == FieldDate)
+  {
+    const CDateTime dt(GetDateTime());
+    const std::string strDt(dt.IsValid() ? dt.GetAsDBDateTime() : "");
+    sortable[FieldDate] = StringUtils::Format("%s %08u", strDt.c_str(), m_index);
```

Comment:
Why `8` digits`? 2^32 - 1 = 4294967295 and that's 10 digits. Obviously we're not gonna have that many events in one session but we should do things properly anyway.